Error: Invalid index
│
│   on main-disk-attachment.tf line 16, in resource "azurerm_virtual_machine_data_disk_attachment" "disk_attachment":
│   16:   managed_disk_id           = module.compute-managed_disk.managed_disk_list[each.value.disk].id
│     ├────────────────
│     │ each.value.disk is "disk-data"
│     │ module.compute-managed_disk.managed_disk_list is object with 6 attributes
│
│ The given key does not identify an element in this collection value.
╵
╷
│ Error: Invalid index
│
│   on main-disk-attachment.tf line 16, in resource "azurerm_virtual_machine_data_disk_attachment" "disk_attachment":
│   16:   managed_disk_id           = module.compute-managed_disk.managed_disk_list[each.value.disk].id
│     ├────────────────
│     │ each.value.disk is "disk-data"
│     │ module.compute-managed_disk.managed_disk_list is object with 6 attributes
│
│ The given key does not identify an element in this collection value.
